lesson-4/README.md

```diff
@@ -0,0 +1,234 @@
+        - --config.file=/etc/prometheus/prometheus.yml
+        - --storage.tsdb.path=/prometheus
+        - --storage.tsdb.retention=24h
+        command:
+        - /bin/prometheus
+        image: quay.io/prometheus/prometheus:v2.3.1
+        imagePullPolicy: IfNotPresent
+        name: prometheus
+        ports:
+        - containerPort: 9090
+          protocol: TCP
+        resources:
+          limits:
+            cpu: 500m
+            memory: 2500Mi
+          requests:
+            cpu: 100m
+            memory: 100Mi
+        volumeMounts:
+        - mountPath: /prometheus
+          name: data
+        - mountPath: /etc/prometheus
+          name: config-volume
+      restartPolicy: Always
+      securityContext: {}
+      terminationGracePeriodSeconds: 30
+      volumes:
+      - emptyDir: {}
+        name: data
+      - configMap:
+          name: prometheus-config
+        name: config-volume
+```
+
+In the *metadata section*, we give the pod a label with a key of name and a value of prometheus.
+This will come in handy later.
+
+In *annotations*, we set a couple of key/value pairs that will actually allow Prometheus
+to autodiscover and scrape itself.
+
+We are using an [*emptyDir*](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir)
+volume for the Prometheus data. It exists as long as Prometheus Pod is running on that node.
+As the name says, it is initially empty.
+Containers in the Prometheus Pod can all read and write the same files in the emptyDir volume,
+though that volume can be mounted at the same or different paths in each Container.
+When a Pod is removed from a node for any reason, the data in the *emptyDir* is deleted forever.
+
+To install Prometheus run the following command:
+
+```bash
+kubectl apply -f prometheus-deployment.yaml
+```
+
+
+Make sure that prometheus is up and running:
+```bash
+NAME         DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+prometheus   1         1         1            1           52s
+```
+
+#### Prometheus Service
+In order to get to the UI of Prometheus we have to expose it via [Kubernetes Services](https://kubernetes.io/docs/concepts/services-networking/service/). In our
+workshop we will use [NodePort](https://kubernetes.io/docs/concepts/services-networking/service/#nodeport)
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: prometheus
+  namespace: monitoring
+spec:
+  ports:
+  - port: 9090
+    protocol: TCP
+    targetPort: 9090
+    nodePort: 30090
+  selector:
+    name: prometheus
+  type: NodePort
+```
+
+To add NodePort service to your kubernetes cluster run
+```bash
+kubectl apply -f prometheus-service.yaml
+```
+List all your services in `monitoring` namespace:
+```bash
+kubectl get svc -n monitoring
+NAME         TYPE       CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
+prometheus   NodePort   10.102.142.14   <none>        9090:30090/TCP   9s
+```
+
+Now you can open prometheus UI via:
+```
+minikube service prometheus -n monitoring
```

Comment:
I have just an open question here, wouldn't it be more educative to use `kubectl proxy ...` here? 
@dbecvarik @Evalle @tomkukral

Reply:
`kubectl proxy` doesn't work with Prometheus UI properly because they are expecting to run in root path. There is some problem with js code in `/js`. 

Only way to use UI properly `kubectl port-forward`.

Reply:
Ingress should work too, right?

Reply:
But this is only cosmetic, so lets use port-forward and add warning to a text that it cannot be used with kube-proxy now? I think its very good info.

Reply:
I'll add it in next PR.